### PR TITLE
[setup] make fairscale and deepspeed setup extras

### PR DESCRIPTION
**Pull request by stas00**
Based on a request adding support for:
```
pip install transformers[deepspeed]
pip install transformers[fairscale]
```
so moving the version minimums into `setup.py`, and also had to add a helper function `dep_version_check`

@LysandreJik, @sgugger

### CODE REVIEW

File: docs/source/main_classes/trainer.rst [hunk 1]
```diff
@@ -272,7 +272,15 @@ Install the library via pypi:
...
 .. code-block:: bash
 
-    pip install fairscale
+   pip install fairscale
+
+or via ``transformers``' ``extras``:
+
+.. code-block:: bash
+
+   pip install transformers[fairscale]
```
Comment (sgugger) [inline docs/source/main_classes/trainer.rst:281]: Let's keep the indentation at four spaces.
```suggestion
    pip install fairscale

or via ``transformers``' ``extras``:

.. code-block:: bash

    pip install transformers[fairscale]
```

File: docs/source/main_classes/trainer.rst [hunk 2]
```diff
@@ -417,7 +425,15 @@ Install the library via pypi:
...
 .. code-block:: bash
 
-    pip install deepspeed
+   pip install deepspeed
+
+or via ``transformers``' ``extras``:
+
+.. code-block:: bash
+
+   pip install transformers[deepspeed]
```
Comment (sgugger) [inline docs/source/main_classes/trainer.rst:434]:
```suggestion
    pip install deepspeed

or via ``transformers``' ``extras``:

.. code-block:: bash

    pip install transformers[deepspeed]
```

**Review by sgugger — APPROVED** [review]
Nice addition!

**Review by LysandreJik — APPROVED** [review]
Great, LGTM

**Pull request merged**